### PR TITLE
Case list distance sort

### PR DESCRIPTION
**Pull request by emord**
@dannyroberts @forest-tong

checked it out on mobile and sorting worked

cc @sravfeyn

### CODE REVIEW

File: corehq/apps/app_manager/detail_screen.py [hunk 1]
```diff
@@ -275,6 +277,13 @@ class TimeAgo(FormattedDetailColumn):
     SORT_XPATH_FUNCTION = u"{xpath}"
 
 
+@register_format_type('distance')
+class Distance(FormattedDetailColumn):
+    XPATH_FUNCTION = u"if(here() = '', '', if({xpath} = '', '', concat(round(distance({xpath}, here()) div 1000), ' km')))"
```
Comment (dimagimon) [inline corehq/apps/app_manager/detail_screen.py]: E501 line too long (123 > 115 characters)

File: corehq/apps/app_manager/tests/test_case_detail_distance.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            <partial>
+                <field>
+                    <header>
+                        <text>
+                            <locale id="m0.case_short.case_name_1.header"/>
+                        </text>
+                    </header>
+                    <template>
+                        <text>
+                            <xpath function="if(here() = '', '', if(case_name = '', '', concat(round(distance(case_name, here()) div 1000), ' km')))"/>
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_case_detail_distance.py]: E501 line too long (151 > 115 characters)

File: corehq/apps/app_manager/tests/test_case_detail_distance.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            <partial>
+                <field>
+                    <header>
+                        <text>
+                            <locale id="m0.case_long.case_name_1.header"/>
+                        </text>
+                    </header>
+                    <template>
+                        <text>
+                            <xpath function="if(here() = '', '', if(case_name = '', '', concat(round(distance(case_name, here()) div 1000), ' km')))"/>
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_case_detail_distance.py]: E501 line too long (151 > 115 characters)

**Comment by dannyroberts** [comment]
Nice! Super cool feature too :) :earth_africa:

File: corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js [hunk 4]
```diff
@@ -1218,6 +1219,12 @@ var DetailScreenConfig = (function () {
             {value: "calculate", label: DetailScreenConfig.message.CALC_XPATH_FORMAT + ' (Preview!)'}
         );
     }
+
+    if (window.toggles.CASE_LIST_DISTANCE_SORT) {
```
Comment (benrudolph) [inline corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js]: :star2: did not realize we had a global for our `toggles`

File: corehq/apps/app_manager/templates/app_manager/module_view.html [hunk 5]
```diff
@@ -17,6 +17,7 @@
     window.feature_previews.ENUM_IMAGE = {{ request|feature_preview_enabled:"ENUM_IMAGE"|BOOL }};
     window.feature_previews.CALC_XPATHS = {{ request|feature_preview_enabled:"CALC_XPATHS"|BOOL }};
     window.toggles.MM_CASE_PROPERTIES = {{ request|toggle_enabled:"MM_CASE_PROPERTIES"|BOOL }};
+    window.toggles.CASE_LIST_DISTANCE_SORT = {{ request|toggle_enabled:"CASE_LIST_DISTANCE_SORT"|BOOL }};
```
Comment (benrudolph) [inline corehq/apps/app_manager/templates/app_manager/module_view.html]: ah... we don't :disappointed:

Comment (benrudolph) [inline corehq/apps/app_manager/templates/app_manager/module_view.html]: it'd be amazing to have this in `base.html`. (just thinking out load, don't think that's at all in scope)

**Comment by emord** [comment]
:white_check_mark: (would be nice to get this in for @forest-tong demo tomorrow)

**Pull request merged**